selfdrive/ui/qt/onroad.cc

```diff
@@ -256,14 +255,110 @@ void NvgWindow::initializeGL() {
   qInfo() << "OpenGL renderer:" << QString((const char*)glGetString(GL_RENDERER));
   qInfo() << "OpenGL language version:" << QString((const char*)glGetString(GL_SHADING_LANGUAGE_VERSION));
 
-  ui_nvg_init(&QUIState::ui_state);
   prev_draw_t = millis_since_boot();
   setBackgroundColor(bg_colors[STATUS_DISENGAGED]);
 }
 
+void NvgWindow::updateFrameMat(int w, int h) {
```

Comment:
I was thinking we get rid of the NvgWindow and make the path/lane lines their own widget. Then the stack can be: CameraViewWidget -> lane lines / path -> HUD -> alerts.

Reply:
I have tried,  `drawLaneLines` takes 2-4ms to draw in it's own widget, but only takes 0.2-0.4ms  in NvgWindow(10x faster). I suggest to still put it in NvgWindow for now. it will be easy to move out to it's own widget after the performance is not an issue.

Reply:
why is it slower?

Reply:
It should be caused by the performance of anti-aliasing. there is no  MSAA in normal widget.

Reply:
Nanovg use MSAA or a shader (if flag NVG_ANTIALIAS is set) to do anti-aliasing.

Reply:
ok, let's merge this for now, but I'd love to separate these out at some point